File: Backend/src/main/java/com/TrainingZone/Server/Repositories/UserRepository.java
package com.TrainingZone.Server.Repositories;

import com.TrainingZone.Server.Models.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User,String> {
    Optional<User> findByEmailOrPhone(String email, String phone);
}

File: Backend/TrainingZone/TrainingZone/Repositories/UserRepository.cs
using Microsoft.EntityFrameworkCore;
using TrainingZone.Models.DataBase;
using TrainingZone.Models.Enums;
using TrainingZone.Repositories.Base;

namespace TrainingZone.Repositories;

public class UserRepository: Repository<User,int>
{
    public UserRepository(TrainingZoneContext context) : base(context) { }

    public async Task<User> GetUserByCredentialAsync(string credential)
    {
        return await GetQueryable()
            .FirstOrDefaultAsync(user => user.Email == credential || user.Phone == credential);
    }

    public async Task<List<User>> GetAllTrainersAsync()
    {
        return await GetQueryable()
            .Where(user => user.Role == Role.TRAINER.ToString().ToLower())
            .ToListAsync();
    }

}
